### ISSUE TITLE
SeparableConv2D from tf.keras.layers and tf.layers gives different results

### ISSUE THREAD

**Issue by hsm207** [tensorflow/tensorflow#18338]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7.0
- **Python version**: 3.6.5
- **Bazel version (if compiling from source)**: NA
- **GCC/Compiler version (if compiling from source)**: NA
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: NA
- **Exact command to reproduce**: See source code

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

The shape of the weight matrices of a SeparableConv2D using tf.keras.layers and tf.layers is different even when using the same arguments. It looks like the SeparableConv2D class from tf.keras.layers is ignoring the `depth_multiplier` argument.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

```
import tensorflow as tf

sess = tf.InteractiveSession()

sep_conv1 = tf.keras.layers.SeparableConv2D(filters=2, kernel_size=2, strides=1, padding='same', depth_multiplier=5)
sep_conv2 = tf.layers.SeparableConv2D(filters=2, kernel_size=2, strides=1, padding='same', depth_multiplier=5)

x = tf.random_uniform((1, 4, 4, 3), minval=1, maxval=10, dtype=tf.int32)
x = tf.cast(x, tf.float32)

y1 = sep_conv1(x)
y2 = sep_conv2(x)

init = tf.global_variables_initializer()
sess.run(init)
sess.run([y1, y2])

# the shape of the depthwise_kernel is (2, 2, 3, 1) and the shape of the pointwise_kernel is (1, 1, 3, 2)
for w in sep_conv1.variables:
    print(w)

# the shape of the depthwise_kernel is (2, 2, 3, 5) and the shape of the pointwise_kernel is (1, 1, 15, 2)
for w in sep_conv2.variables:
    print(w)
```

**Comment by kr-ish** [comment]
You're right, seems the keras implementation takes the depth_multiplier argument but does nothing with it. This seems to be true for both SeparableConv1D and SeparableConv2D. See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/_impl/keras/layers/convolutional.py#L941-L983. I'd be happy to put in a PR fixing it but I wonder if it was left out intentionally?

**Comment by kr-ish** [comment]
I think a fix would look something like this- https://github.com/kr-ish/tensorflow/commit/676778f16e6efbde5b59f753e7e85d74359d046f, but perhaps with an added test case

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 106 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 121 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 136 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 151 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 166 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 181 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 196 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 211 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 226 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
It is still replicating in[ 2.8 ](https://colab.sandbox.google.com/gist/mohantym/0f0d2e5fc3f6f9f754bc237c6c203c6a/github_18338.ipynb)version.
